File: temperatures.py
import sys
import math

n = int(input())  # the number of temperatures to analyse
result = 0
temperature: int = None
for i in input().split():
    # t: a temperature expressed as an integer ranging from -273 to 5526
    t = int(i)
    if (temperature is None or abs(t) < abs(temperature) or t == abs(temperature)):
        temperature = t
        result = i

print(str(result))

File: temperatures.swift
import Glibc
import Foundation

public struct StderrOutputStream: TextOutputStream {
  public mutating func write(_ string: String) { fputs(string, stderr) }
}
public var errStream = StderrOutputStream()

let n = Int(readLine()!)! // the number of temperatures to analyse
var result: String?
var temperature: Int?
for i in ((readLine()!).split(separator: " ").map(String.init)) {
  let t = Int(i)!
    
  if temperature == nil || abs(t) < temperature.map(abs)! || t == temperature.map(abs) {
    temperature = t
    result = i    
  }
}

print("\(result ?? "0")")
